### DISCUSSION TITLE
Difference between sender container in From header and X-Simplelogin-Envelope-From

### DISCUSSION THREAD

**Discussion by MPeti1** [simple-login/app#1470] [Q&A]
Recently I've received an email which was marked as suspicious by SL's anti-phising filter, and while investigating it, I've found something I don't understand.

Based on the timing of certain events, I'm fairly certain that this is a legit email.
I wanted to let the website owner know that they might have misconfigured something, so I have dived in the email headers to see if I can provide useful details.
Among the email headers, I see 2 different senders mentioned:
- The header with name `X-Simplelogin-Envelope-From` says this email has been sent from the website owner's own email address, with a domain of their own website
- The headers with name `From` and `Reply-To` mention an other email address: this one has the domain of gmail.com, and a slightly different username part.

The alias log of the relevant alias (at `https://app.simplelogin.io/dashboard/alias_log/<alias id>`) shows the gmail address for the relevant forwarded email. Here in the app.simplelogin.io web client I don't see the email address mentioned in the `X-Simplelogin-Envelope-From` header, but of course I might have just missed it.

What is the meaning of `X-Simplelogin-Envelope-From`, and what address is put into `From`?
Practically what is the difference between the 2 sender indications? In what circumstances can they be different?

**Comment by nguyenkims** [comment]
@MPeti1 The `X-Simplelogin-Envelope-From` is the original **envelope** mail_from. Please note this isn't the same as the from header which is displayed in the email client.
The `From` header is converted into a reverse alias during the forwarding so when you reply, the response will be sent to the reverse alias which will then the email from your alias.
